src/Akka.Streams.Kafka/Stages/Consumers/Abstract/BaseSingleSourceLogic.cs

```diff
@@ -123,15 +128,28 @@ protected virtual void MessageHandling((IActorRef, object) args)
                 
                 case Status.Failure failure:
                     var exception = failure.Cause;
-                    if (_decider(exception) == Directive.Stop)
+                    var cause = exception is KafkaException ke ? ke.Error.Reason : exception.Message; 
+                    var directive = _decider(exception); 
+                    if (directive == Directive.Stop)
                     {
+                        if(Log.IsErrorEnabled)
+                            Log.Error(exception, "Source stage failed with exception: [{0}]. Decider directive: {1}", cause, directive);
                         FailStage(failure.Cause);
                         break;
-                    } 
+                    }
+
+                    if(Log.IsInfoEnabled)
+                        Log.Info(exception, "Source stage failure [{0}] handled with Supervision Directive [{1}]", cause, directive);
                     
                     var isSerializationError = exception is ConsumeException cEx && cEx.Error.IsSerializationError();
                     if (isSerializationError)
-                        break;
+                    {
+                        if (directive == Directive.Resume)
```

Comment:
Need to make sure that the user have an option to restart when serialization error occured